### PR TITLE
feat: add the plugin:vitest/recommended in extends array

### LINKED ISSUE #809: 🚀 Feature: Extend from plugin:vitest/recommended in .eslintrc.cjs
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

https://github.com/veritem/eslint-plugin-vitest#recommended isn't yet enabled in this template. Let's turn it on in the ESLint config!

### Additional Info

This is halfway to a bug, honestly...

### PR DESCRIPTION
**Pull request by jdwilkin4**
## PR Checklist

- [x] Addresses an existing open issue: fixes #809 
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/create-typescript-app/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

This PR is responsible for adding the `plugin:vitest/recommended` to the extends array in the `.eslintrc.cjs` file. 

This PR also updates a few of the tests because once the `plugin:vitest/recommended` was added, some of the tests failed the lint checks. 

For the following files, I replaced the `toEqual ` methods with the `toBe` methods per the vitest recommendations

- src/bin/index.test.ts
- src/shared/doesRepositoryExist.test.ts
- src/shared/runOrRestore.test.ts
- src/steps/writing/creation/dotGitHub/createWorkflowFile.test.ts

For the following files, I replaced the `toBe(undefined)` with `toBeUndefined()` per the vitest recommendations

- src/shared/options/readGitAndNpmDefaults/readTitleFromReadme.test.ts
- src/shared/tryCatchLazyValueAsync.test.ts

### CODE REVIEW

File: src/steps/initializeBranchProtectionSettings.test.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ const createMockOctokit = (request: SpyInstance) =>
 const stubValues = { owner: "", repository: "" };
 
 describe("migrateBranchProtectionSettings", () => {
+	// eslint-disable-next-line vitest/expect-expect
```
Comment (JoshuaKGoldberg) [inline src/steps/initializeBranchProtectionSettings.test.ts:14]: The lint rule is right though. Per https://vitest.dev/api/expect.html#rejects the right test API is:

```ts
await expect(
	initializeBranchProtectionSettings(
		createMockOctokit(mockRequest),
		stubValues,
	),
).resolves.not.toThrow();
```

Let's go with that?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
✨  Love it! Very close, just one thought on using a Vitest API instead of a disable comment. Thoughts?

File: src/steps/initializeBranchProtectionSettings.test.ts (hunk 1 above)
Comment (jdwilkin4) [reply src/steps/initializeBranchProtectionSettings.test.ts:14]: yeah, that makes sense. 
I can go ahead and update it 👍

**Review by JoshuaKGoldberg — APPROVED** [review]
💯 thanks!

**Pull request merged**